modules/tooling-support-parent/tooling-support/src/main/java/org/mule/runtime/module/tooling/api/metadata/ComponentMetadataTypes.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.tooling.api.metadata;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
+
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.runtime.api.metadata.descriptor.InputMetadataDescriptor;
+import org.mule.runtime.api.metadata.descriptor.OutputMetadataDescriptor;
+import org.mule.runtime.api.metadata.descriptor.ParameterMetadataDescriptor;
+import org.mule.runtime.api.metadata.descriptor.TypeMetadataDescriptor;
+import org.mule.runtime.api.metadata.resolving.MetadataResult;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Exposes all the dynamic metadata (input, output, attributes) of a given component.
+ */
+public class ComponentMetadataTypes {
+
+  private Map<String, MetadataType> input = new HashMap<>();
+  private MetadataType output;
+  private MetadataType outputAttributes;
+
+  public ComponentMetadataTypes(MetadataResult<InputMetadataDescriptor> inputMetadataResult,
```

Comment:
I would move this to a "builder" as this object would be used as POJO for sending the response to the client.

Reply:
Ok, I'll move the constructor logic to a builder.
Any requirements by the serializer regarding constructors? like having a public no-args one?

Reply:
Not so far but I would need to move this class lo mule-api. Let's keep it here for now and I will change it in next PR when adding serialization support.